### PR TITLE
Improve functest

### PR DESCRIPTION
**Pull request by chanchiwai-ray**
Add basic functional testing for the charm.

### CODE REVIEW

File: src/charm.py [hunk 1]
```diff
@@ -25,25 +26,33 @@ class PrometheusHardwareExporterCharm(ops.CharmBase):
...
+        self.framework.observe(
+            self.on.cos_agent_relation_departed, self._on_cos_agent_relation_departed
+        )
+
         self._stored.set_default(installed=False, config={})
 
     def _on_install_or_upgrade(self, _: EventBase) -> None:
         """Install and upgrade."""
-        self.exporter.install()
+        self.exporter.install(dict(self.model.config))
```
Comment (jneo8) [inline src/charm.py:55]: **nitpick:** 

It's not useful in the future if we have more configuration that is not required by exporter.

File: tests/functional/conftest.py [hunk 2]
```diff
@@ -5,6 +5,16 @@
 log = logging.getLogger(__name__)
 
 
+class Helper:
```
Comment (jneo8) [inline tests/functional/conftest.py:8]: **nitpick:** 

helper is too generic.

**Review by jneo8 — CHANGES_REQUESTED** [review]

File: tests/functional/conftest.py (hunk 2 above)
Comment (chanchiwai-ray) [reply tests/functional/conftest.py]: DONE

File: src/charm.py (hunk 1 above)
Comment (chanchiwai-ray) [reply src/charm.py]: DONE

**Review by Pjack — APPROVED** [review]

**Review by jneo8 — APPROVED** [review]
Wait for functional test pass.

**Review by sudeephb — APPROVED** [review]

**Comment by chanchiwai-ray** [comment]
Not sure why it's failing on github runner; test on local lxd cloud works https://pastebin.ubuntu.com/p/gTgXSwk563/

**Pull request merged**